trihoang0809/tourific
# Friend-related Routes

# Description
- Implement logic for adding a friend, view pending/sent friend requests, accept/decline a friend request, get all friends

## Issues

## Screenshots
![Screenshot 2024-06-06 111139](https://github.com/trihoang0809/tourific/assets/89886131/ebb4e4f9-8eda-4858-8366-9da769a01502)
![Screenshot 2024-06-06 111212](https://github.com/trihoang0809/tourific/assets/89886131/3ca7d5ef-5eef-47ab-86c6-11a7059cbc18)
![Screenshot 2024-06-06 111603](https://github.com/trihoang0809/tourific/assets/89886131/3c237f60-166c-413b-8c2c-c6c283e1b281)
![Screenshot 2024-06-06 111938](https://github.com/trihoang0809/tourific/assets/89886131/fe2a0d09-f4ee-4156-b842-b8bb994cf16b)
![Screenshot 2024-06-06 112021](https://github.com/trihoang0809/tourific/assets/89886131/532b4992-95eb-49d9-8548-8eb2afe0d1d1)


## Test
- Try create users or view user's list in database
- Pick a random user and change userId in UserProfile.tsx to the chosen user
- Try send a friend request from that user and check route /friend/sent-requests to see if the requests showing up. Then save id for both sender and receiver
- To reject/accept a user, from UserProfile.tsx change userId to receiverId from the step above

## Possible Downsides
- Missing route to view all rejected friend request (I don't think it's necessary for now)

## Additional Documentations
- Route starts with http://localhost:3000/user/
